clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java

```diff
@@ -61,7 +61,7 @@ public static void registerAppInfo(String prefix, String id) {
         }
     }
 
-    public static void unregisterAppInfo(String prefix, String id) {
+    public static synchronized void unregisterAppInfo(String prefix, String id) {
```

Comment:
yes for consistency, we can synchronize registerAppInfo method also. when multiple clients tries register mbean with the same clientId, we print an error message and continue.  when multiple threads try to unregister we delete only if mbean exist in the mbean registry.

updated the PR.